### PR TITLE
Allows to suppress "caused by" in error message, not only the stacks

### PR DESCRIPTION
**Pull request by dhs3000**
Currently the error handler allows to suppress stack traces in the generated output. However, to easier circumvent [owasp | Improper Error Handling](https://owasp.org/www-community/Improper_Error_Handling) issues, I think it would be good to also suppress the "caused by" that includes the full exception class name.

(Currently I need to overwrite `ErrorHandler#generateResponse` and pass `null` for the `cause` parameter to achieve this.)

### CODE REVIEW

**Review by gregw — DISMISSED** [review]
LGTM.  Just rebuilding to see if the CI failure was a flake

**Comment by dhs3000** [comment]
Forgot to send my comment earlier: As the default for `showStacks` is switched to `false` now, potentially the same should be done for the new `showCauses`?

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java [hunk 1]
```diff
@@ -71,6 +71,7 @@ public class ErrorHandler implements Request.Handler
     public static final HttpField ERROR_CACHE_CONTROL = new PreEncodedHttpField(HttpHeader.CACHE_CONTROL, "must-revalidate,no-cache,no-store");
 
     boolean _showStacks = false;
+    boolean _showCauses = true;
```
Comment (sbordet) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java]: Please make it default to `false`.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java (hunk 1 above)
Comment (dhs3000) [reply jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java]: I switched the default to false

**Comment by dhs3000** [comment]
@gregw @sbordet I updated the PR switching the default to false (going along the default for `showStacks`)

**Comment by gregw** [comment]
You have a checkstyle failure:
```
[INFO] Starting audit...

[WARN] /home/jenkins/agent/workspace/jetty.project_PR-11552/jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ErrorHandlerTest.java:523:5: 'METHOD_DEF' has more than 1 empty lines before. [EmptyLineSeparator]

Audit done.

[WARNING] src/test/java/org/eclipse/jetty/server/ErrorHandlerTest.java:[523,5] (whitespace) EmptyLineSeparator: 'METHOD_DEF' has more than 1 empty lines before.
```

**Comment by joakime** [comment]
@dhs3000 you still have checkstyle issues.

Try this quick check first.

```
mvn checkstyle:check
```

But due to the build cache, it might not find anything, so in that case use this ...

```
mvn clean install -Dmaven.build.cache.enabled=false -DskipTests -Pci 
```

This is a full build, without testing, using the same rules as CI.
It will show you the checkstyle issues on your command line.

**Comment by sbordet** [comment]
I think we agree that the contribution is good and accepted, but there are test failures that check for the `ErrorHandler` output that is now suppressed by default.

If you don't mind, I'll take over as I think this is now on our court, rather than @dhs3000, who already provided a good PR.

**Comment by dhs3000** [comment]
I ran `mvn clean install -Dmaven.build.cache.enabled=false -DskipTests -Pci ` in `/jetty-core/jetty-server` (where I did changes)
```
[INFO] --- checkstyle:3.3.1:check (checkstyle-check) @ jetty-server ---
[INFO] Starting audit...
Audit done.
[INFO] You have 0 Checkstyle violations.
```

**Comment by dhs3000** [comment]
Thank you @sbordet. (`Allow edits by maintainers` is enabled on the PR, if that helps.)

**Comment by sbordet** [comment]
See #11587.

**Comment by joakime** [comment]
Replaced by PR #11587

**Pull request closed without merging**